### PR TITLE
Add an authorization validation rule

### PR DESCRIPTION
**Pull request by Shane32**

### CODE REVIEW

File: src/Tests/AuthorizationTests.cs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+namespace Tests;
+
+public class AuthorizationTests
+{
+
```
Comment (sungam3r) [inline src/Tests/AuthorizationTests.cs:5]: 😕

Comment (Shane32) [reply src/Tests/AuthorizationTests.cs:5]: Lol no I didn’t finish last night like I was hoping

File: README.md [hunk 2]
```diff
@@ -17,8 +17,12 @@ builder interfaces.
 In addition, an `ExecutionResultActionResult` class is added for returning `ExecutionResult`
 instances directly from a controller action.
 
-You will need to register the middleware and the WebSockets handler in the dependency injection
-framework in order to use them.
+Authorization is also supported with the included `AuthorizationValidationRule`.  It will
+scan GraphQL documents and validate that all referenced output graph types, fields of
```
Comment (Shane32) [inline README.md:21]:
```suggestion
scan GraphQL documents and validate that the schema and all referenced output graph types, fields of
```

File: README.md [hunk 3]
```diff
@@ -17,8 +17,12 @@ builder interfaces.
...
 instances directly from a controller action.
 
-You will need to register the middleware and the WebSockets handler in the dependency injection
-framework in order to use them.
+Authorization is also supported with the included `AuthorizationValidationRule`.  It will
+scan GraphQL documents and validate that the schema and all referenced output graph types, fields of
+output graph types, and query arguments meet the specified policy and/or roles held by the
+authenticated user within the ASP.NET Core authorization framework.  It does not validate
+any policies or roles specified for input graph types, fields of input graph types, or
+directives.
```
Comment (Shane32) [inline README.md:25]:
```suggestion
directives.  It does not skip validation for fields that are marked with the `@skip` or
`@include` directives.
```

**Comment by Shane32** [comment]
@sungam3r I've added 179 tests; everything works as expected.

**Comment by Shane32** [comment]
Test coverage has gone down a little bit due to a lot of conditionals in case of invalid documents being processed.

**Comment by Shane32** [comment]
@sungam3r I added api approvals so at least you can see all of the API in one place.  We can of course make changes before I release 1.1.0 here.  Test coverage should be fairly extensive ... I hope ... but it's tough to get everything.  And I didn't write tests for documents that won't pass the core set of rules.  (That could be valuable, to be sure a null reference exception doesn't occur or something.)  Note that validation is skipped for nodes that GraphQL can't match to fields.

**Pull request merged**